addons/mail/static/src/models/notification_group/notification_group.js

```diff
@@ -67,6 +67,14 @@ function factory(dependencies) {
             return `${this.modelName}_${data.id}`;
         }
 
+        _computeDate() {
+            return moment.max(this.notifications.map((notification) => notification.message.date));
+        }
```

Comment:
As you did in the PR, `message.date` might be undefined now, and so could be `group.date`.